File: Week 1/Python Codes/ProductArrayPuzzle.py
def product_array(n, arr):
    product = 1
    product_array = [1]*n
    for num in arr:
        product = product * num
    for i in range(n):
        product_array[i] = product // arr[i]
    for num in product_array:
        print(num, end = " ")
    print()

if __name__ == '__main__':
    t = int(input())
    for i in range(t):
        n = int(input())
        arr = [int(x) for x in input().split()]
        product_array(n, arr)  

File: Week 1/C++ Codes/ProductArrayPuzzle.cpp
#include <iostream>
using namespace std;

void product_array(int n, int arr[]) {
    long long int product = 1;
    int product_array[n];
    for (int i=0; i<n; i++) {
        product = product * arr[i];
    }
    for (int i=0; i<n; i++) {
        product_array[i] = product / arr[i];
        cout<<product_array[i]<<" ";
    }
    cout<<"\n";
}

int main() {
	//code
	int t, n;
	cin>>t;
	while (t--) {
	    cin>>n;
	    int arr[n];
	    for(int i=0; i<n; i++)
	        cin>>arr[i];
	    product_array(n, arr);
	}
	return 0;
}
